serverlessworkflow/modules/ROOT/pages/cloud/kubernetes-service-discovery.adoc

```diff
@@ -24,90 +22,93 @@ kubernetes:<group>/<version>/<kind>/<namespace>/<resourceName>?<attributeName>=<
-** `labels=label-name=namevalue;another-label=another-value` +
-    Example:
+* *Custom labels*: The custom labels are used to filter services in case there are more than one service with the same label selector but exposing different ports. In this case, you can instruct the engine that if more than one service is found, then the engine must use the service containing the provided label. 
++
+The label is defined with the following expression and in case of multiple labels, you can use semicolon (;):
++
+`labels=label-name=namevalue;another-label=another-value`
 +
+.Example label definition in URI
 [source,shell]
 ----
 kubernetes:v1/pod/<namespace>/<pod-name>?labels=label-name=test-label
 ----
-*** With the URI above we will instruct that, if there are more than 1 service exposing the given pod, the label
-`label-name=test-label` will be used to filter the service. If the label does not exist, the first found service will be used.
++
+Using the previous URI example, if there are more than one service exposing the given pod, the `label-name=test-label` label is used to filter the service. If the label does not exist, the first found service is used.
+
+* *Custom port name*: The custom port name is used to determine which port to use when multiple ports are configured in the target service or container. You can configure the port name to be queried using the following pattern:
++
+`port-name=<PORT_NAME>`
+--
 
-* **Custom port name**: Useful to determine which port to use when there are many ports configured in the target Service or Container.
-It can be configured using the following patter:
-** `port-name=<PORT_NAME>`: the port name to be queried.
+Based on the resource to be discovered, the Kubernetes service discovery follows specific paths as shown in the following figure:
 
+image::cloud/sw-discovery-flow.jpg[]
 
-== Configuration
+There is no specific configuration required for the Kubernetes service discovery except by using the expected URI pattern. However, the `okhttp` communication interceptor, which logs the communication between the application and the Kubernetes API is disabled by default.
 
-There is no specific configuration for the Discovery to work properly except by using the expected URI pattern.
-However, the okhttp communication interceptor, which logs the communication between the application and the Kubernetes API is disabled by default. +
-If there is a need to debug it, it can be enabled just by setting this application property:
+You can enable the `okhttp` communication interceptor if there is a need to debug it by setting the following application property:
 
+.Property to enable `okhttp` communication interceptor
 [source,shell]
 ----
 quarkus.log.category."okhttp3.OkHttpClient".level=INFO
 ----
 
+[[ref-example-kubernetes-service-discovery]]
+== Example of Kubernetes service discovery in {context}
```

Comment:
which example are you referring to?

Reply:
referring to the `serverless-workflow-greeting-quarkus` example, which is also used in the URI in the section.

Reply:
not sure if I understand it.

Reply:
We have a URI in this section as follows which contains `serverless-workflow-greeting-quarkus`, therefore added example:

![image](https://user-images.githubusercontent.com/59050394/190379037-96714827-b0ce-4aac-a156-530d0bc17233.png)